src/main/webapp/script.js

```diff
@@ -0,0 +1,232 @@
+  // check file size.
+  validateImageSize(file, fileType);
+
+  // Image resolution can not be more than 1920x1080.
+  await validateImageResolution(file);
+}
+
+/**
+ * Function to make sure the image is not too big.
+ * @param {File} image
+ * @param {string} imageType Can be 'png' or 'jpeg' only.
+ */
+function validateImageSize(image, imageType) {
+  // This limit comes from simple calculation.
+  // MAX_RESOLUTION * AVERAGE_NUMBER_OF_BYTES_PER_PIXEL_IN_PNG
+  // 1920 x 1080 x 4 ~ 8294400 bytes ~ 8 mb
+  const PNG_LIMIT_MB = 8;
+  if (imageType === 'png' && image.size > PNG_LIMIT_MB * 1024 * 1024) {
+    throw new Error(
+        `File size should not exceed ${PNG_LIMIT_MB}MB for png images.` +
+        'The size of an uploaded png image is ' +
+        Math.ceil(image.size / 1024 / 1024) + 'MB');
+  }
+
+  // This limit comes from similar calculation.
+  // MAX_RESOLUTION * AVERAGE_NUMBER_OF_BITS_PER_PIXEL_IN_JPG
+  // 1920 x 1080 x 8.25 ~ 17107200 bits ~ 2 mb
+  const JPG_LIMIT_MB = 2;
+  if (imageType === 'jpeg' && image.size > JPG_LIMIT_MB * 1024 * 1024) {
+    throw new Error(
+        `File size should not exceed ${JPG_LIMIT_MB}MB for jpeg images.` +
+        'The size of an uploaded jpeg image is ' +
+        Math.ceil(image.size / 1024 / 1024) + 'MB');
+  }
+}
+
+/**
+ * Function to get file type if it is png or jpeg
+ * and throw error otherwise
+ * @param {File} file
+ * @return {Promise}
+ */
+function getImageTypeOrError(file) {
+  return new Promise(function(resolve, reject) {
+    const fileReader = new FileReader();
+
+    fileReader.onloadend = function(e) {
+      // get first 4 bytes from file - they contain information
+      // about the extension.
+      const bytes = new Uint8Array(e.target.result);
+
+      // convert those bytes to string
+      let header = '';
+      for (let i = 0; i < bytes.length; i++) {
+        header += bytes[i].toString(16);
+      }
+
+      const PNG_HEADERS = ['89504e47'];
+      if (PNG_HEADERS.includes(header)) {
+        resolve('png');
+      }
+
+      const JPEG_HEADERS = ['ffd8ffe0', 'ffd8ffe1',
+        'ffd8ffe2', 'ffd8ffe3', 'ffd8ffe8'];
+      if (JPEG_HEADERS.includes(header)) {
+        resolve('jpeg');
+      }
+
+      reject(new Error(
+          'Invalid file type. Only jpeg and png images can be uploaded'));
+    };
+
+    // read first 4 bytes from file
+    fileReader.readAsArrayBuffer(file.slice(0, 4));
+  });
+}
+
+/**
+ * Function to make sure the image resolution is no
+ * more than 1920 x 1080 (Full HD)
+ * Promise returns error if this is not true and
+ * nothing otherwise.
+ * @param {File} imageFile
+ * @return {Promise}
+ */
+function validateImageResolution(imageFile) {
+  return new Promise(function(resolve, reject) {
+    // construct image object from the file
+    const imageObject = new Image();
+    const imageUrl = URL.createObjectURL(imageFile);
+
+    imageObject.onload = function() {
+      if (imageObject.width * imageObject.height > 1920 * 1080) {
```

Comment:
Same with numbers, 1920 and 1080. It's good to have a single source of truth, and just referencing that everytime you use them to maintain consistency.

Reply:
In this case I agree that it'd better to put these numbers into parameters of the function - at least for the purpose of not changing the error message below if the max resolution changes. There is a number of things relying on these numbers - the max size of the pictures, the error message in the tests. Ideally we'd need to change them as well.